game/src/Game/GameSession/billboard.gdshader

```diff
@@ -0,0 +1,39 @@
+shader_type spatial;
+render_mode unshaded;
+
+//vic2 only ever loads a max of 12 BillboardType
+//Of these, only 4 are actually used
+uniform sampler2D billboards[12] : source_color;
+uniform int numframes[12];
+uniform float sizes[12];
+
+uniform float height_add_factor = 0.3;
+
+//COLOR/INSTANCE_CUSTOM is our custom data, used as follows:
+// x=image index
+// y=frame in image index
+// z,w are unused for now
+
+void vertex() {
+	COLOR = INSTANCE_CUSTOM; //send instance_custom info to fragment
+	float size = sizes[int(COLOR.x)];
+	
+	vec3 cam_right_worldspace = vec3(VIEW_MATRIX[0][0],VIEW_MATRIX[1][0],VIEW_MATRIX[2][0]);
+	vec3 cam_up_worldspace = vec3(VIEW_MATRIX[0][1],VIEW_MATRIX[1][1],VIEW_MATRIX[2][1]);
+	vec3 vert_pos_world =
+		cam_right_worldspace * VERTEX.x * size +
+		cam_up_worldspace * VERTEX.y * size;
+	VERTEX = vert_pos_world + vec3(0,height_add_factor*size,0);
+}
+
+void fragment() {
+	int image_index = int(COLOR.x);
+	//Color.y == 0 >> invisible, otherwise use the specified frame
+	float visibility = clamp(COLOR.y,0.0,1.0);
+	float frame_index = max(COLOR.y,0.0);
+
+	float uv_x_space = 1.0/float(numframes[image_index]);
+	vec2 uv_out = vec2(uv_x_space * (frame_index + UV.x),UV.y);
+	ALBEDO.rgb = texture( billboards[image_index] ,uv_out).rgb;
+	ALPHA = texture(billboards[image_index] ,uv_out).a * visibility;
```

Comment:
```suggestion
	vec4 billboard_col = texture( billboards[image_index] ,uv_out);
	ALBEDO.rgb = billboard_col.rgb;
	ALPHA = billboard_col.a * visibility;
```
Better to look up the pixel once, store it in a variable, and use the values from there, rather than looking it up twice